src/plugins/interactive_setup/common/elasticsearch_connection_status.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
```

Comment:
__note:__ names of the enum members TBD, the idea is to have 3 states:

* We don't know the status yet, we're trying to connect to ES, but it hasn't responded yet. It can be just slow connection, or server isn't reachable, whatever.
* We know that we could successfully connect to ES
* We know that we couldn't connect to ES